operator/src/main/java/oracle/kubernetes/operator/helpers/PodStepContext.java

```diff
@@ -444,6 +515,77 @@ public NextAction onSuccess(Packet packet, CallResponse<V1Pod> callResponse) {
     }
   }
 
+  Step verifyPersistentVolume(Step next) {
+    return new VerifyPersistentVolumeStep(next);
+  }
+
+  private class VerifyPersistentVolumeStep extends Step {
+
+    VerifyPersistentVolumeStep(Step next) {
+      super(next);
+    }
+
+    @Override
+    public NextAction apply(Packet packet) {
+      String domainUID = getDomainUID();
+      Step list =
+          new CallBuilder()
+              .withLabelSelectors(forDomainUid(domainUID))
+              .listPersistentVolumeAsync(
+                  new ResponseStep<V1PersistentVolumeList>(getNext()) {
+                    @Override
+                    public NextAction onFailure(
```

Comment:
Use DefaultResponseStep to get this onFailure behavior without adding code.

Also, consider pulling anonymous inner classes out into named classes, to keep methods short and easier for the next person to read and modify, and please use the newer version of the onSuccess method that includes CallResponse as a replacement for the deprecated version that takes result plus statusCode plus responseHeaders.